### PR TITLE
Join existing shopping list

### LINKED ISSUE #5: 5. As a user, I want to join an existing shopping list so I can share a shopping list with another person
## Summary
We want to allow users to join existing shopping lists so they can manage them with friends or family. This would take the following steps:
1. User enters an existing token into a form
2. The token is saved to `localStorage`
3. The user is taken to the `List` view and shown the existing list.

## Acceptance criteria
If a user **doesn’t** already have a token:
- [x] The `Home` view shows a form that allows the user to enter a token to join an _existing_ list, in addition to the button that allows them to create a _new_ list.
- [x] The input that accepts the list token has a semantic `label` element associated with its
- [x] The user can submit this form with both the mouse and the `Enter` key
- [x] If the list exists,
	- the token is saved in `localStorage`
	- the user is redirected to the `List` view and shown the items on that list
- [x] If the list _does not_ exist, the user is shown an error message that explains the problem

If a user **does** already have a token:
- [x] They are automatically redirected to the `List` view.

## Notes
- Be sure to use the `setListToken` function to save the token that the user enters!
- You will know your `<label>` is correctly implemented if you click on the label and keyboard focus moves to the related input. Refer to [the MDN docs on `<label>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label) for more info.
- Users will be able to submit the form with the `Enter` key if you listen for the right JavaScript event on the right element! If you're adding an `onKeyDown` listener anywhere, you might be overcomplicating things.
- You can assume that users will only be sharing a token if the list has actual items on it. So, let's say I create a New Shopping List, but never add any items to it, and haven't saved that particular token to Firestore, we can assume that's a list that doesn't exist.
- If you would like, you can save the token to Firestore immediately when creating a list, and then a user would be able to share an empty list (see issue 3).

### PR DESCRIPTION
**Pull request by shagunZ**
Co-authored-by: [ToobaJamal](https://github.com/ToobaJamal)

## Description

This change allows users to join an existing shopping list by entering an existing three-word token. If the token is correct, the user is navigated to their shopping list. If a user already has a token in their local storage, they are automatically redirected to the list view.

## Related Issue

Closes #5 

## Acceptance Criteria

If a user doesn’t already have a token:

- [x]  The Home view shows a form that allows the user to enter a token to join an existing list, in addition to the button that allows them to create a new list.

- [x]  The input that accepts the list token has a semantic label element associated with its

- [x]  The user can submit this form with both the mouse and the Enter key

- [x]  If the list exists,

- the token is saved in localStorage

- the user is redirected to the List view and shown the items on that list

- [x]  If the list does not exist, the user is shown an error message that explains the problem

If a user does already have a token:

- [x]  They are automatically redirected to the List view.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before
issue5_before

### After

https://user-images.githubusercontent.com/52610124/232854756-44692b19-8910-4451-b4f1-17b7dbd727e9.mp4

## Testing Steps / QA Criteria

When accessing a non-existing list, the join existing list form throws an error. If the token provided by the user is correct, they are redirected to the list page.

### CODE REVIEW

**Review by labrocadabro — CHANGES_REQUESTED** [review]
Just one thing I think should be changed:

Currently if I want to share my list with someone else, there's no way for me to get the token except to use the dev tools to find it in localStorage. Most users wouldn't know how to do that, so I think the current token should be displayed on the /list page, with a message like "To share this list use the following token:".

File: src/views/Home.css [hunk 1]
```diff
@@ -0,0 +1,6 @@
+form {
+	margin-top: 30px;
+}
+.existing-btn {
+	margin-top: 10px;
+}
```
Comment (paramsiddharth) [inline src/views/Home.css:6]: Point to note: In future we'll need to scope these styles as well as apply some finishing!

**Review by paramsiddharth — APPROVED** [review]
Impressive. Tooba and Shagun! :) Good work!

Although what Laura said makes a lot of sense, the PR passes the acceptance criteria, so I'll be happy to merge this in. If you'd like to make the update Laura suggested, feel free to do that before we merge it in today's sync.

**Comment by RajGM** [comment]
Good work.
 Tooba and Shagun!

**Review by paramsiddharth — APPROVED** [review]

**Pull request merged**